### PR TITLE
For [Doc] Alerting plugin RBAC - backend roles usage 

### PR DESCRIPTION
**Pull request by alicejw-aws**
…site/issues/1810

Signed-off-by: alicejw <alicejw@amazon.com>

### Description
How to use new RBAC role parameter to create a monitor and specify a role.

### Issues Resolved
[Alerting plugin RBAC](https://github.com/opensearch-project/documentation-website/issues/1810)

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _monitoring-plugins/alerting/security.md [hunk 1]
```diff
@@ -82,3 +82,40 @@ If you only want users to be able to see and modify their own monitors and desti
...
+### Specify RBAC backend roles
+
+You can specify RBAC backend roles when you create or update a monitor with the Alerting API.
+
+In a create monitor scenario, follow these guidelines to specify roles:
+
+User type  | Role is specified by user or not (Y/N) | How to use the RBAC roles
+:--- | :--- | :---
+Admin user | User-specified | Use all the specified backend roles to associate to the monitor.
+Regular user | Yes | Use all the specified backend roles (that the user already has in their list of backend roles) to associate to the monitor.
```
Comment (lezzago) [inline _monitoring-plugins/alerting/security.md:95]: For `that the user already has in their list of backend roles`, should we say the list of backend roles that the user has permissions to?

File: _monitoring-plugins/alerting/security.md [hunk 2]
```diff
@@ -82,3 +82,40 @@ If you only want users to be able to see and modify their own monitors and desti
...
+Regular user | No | Copy user’s backend roles and associate them to the monitor.
+
+In an update monitor scenario, follow these guidelines to specify roles:
+
+User type  | Role is specified by user or not (Y/N) | How to use the RBAC roles
+:--- | :--- | :---
+Admin user | User-specified | Remove all the backend roles associate to the monitor and then use all the specified backend roles to associate to the monitor.
+Regular user | Yes | Remove backend roles associated to the monitor that the user has access to, but didn’t specify. Then add all the other specified backend roles (that the user already has in their list of backend roles) to the monitor.
```
Comment (lezzago) [inline _monitoring-plugins/alerting/security.md:103]: Same thing as above

File: _monitoring-plugins/alerting/security.md [hunk 3]
```diff
@@ -82,3 +82,40 @@ If you only want users to be able to see and modify their own monitors and desti
...
+Regular user | No | Don’t update the backend roles on the monitor.
+
+Note that an empty list is considered the same as removing all permissions that the user possesses. Also note that if the user tries to associate roles that they don't have permission to use, it will throw an exception.
```
Comment (lezzago) [inline _monitoring-plugins/alerting/security.md:106]: We need to mention specifying an empty list will only work for admin users.

**Review by lezzago — CHANGES_REQUESTED** [review]

File: _security-plugin/access-control/api.md [hunk 4]
```diff
@@ -427,14 +427,16 @@ Creates or replaces the specified user. You must specify either `password` (plai
 
 Note that any role you supply in the `opendistro_security_roles` array must already exist for the security plugin to map the user to that role. To see predefined roles, refer to [the list of predefined roles]({{site.url}}{{site.baseurl}}/security-plugin/access-control/users-roles#predefined-roles). For instructions on how to create a role, refer to [creating a role](#create-role).
 
+To learn more about backend roles, see [\(Advanced\) Limit access by backend role]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/security/#advanced-limit-access-by-backend-role).
```
[the remark this answers is not in the document]
Comment (alicejw-aws) [reply _security-plugin/access-control/api.md:430]: ok

[the remark this answers is not in the document]
Comment (alicejw-aws) [reply _security-plugin/access-control/api.md:430]: but this is in the Security plugin API page.

**Comment by alicejw-aws** [comment]
thanks @lezzago ! Would you please take another look to verify the changes are correct? best regards

File: _monitoring-plugins/alerting/api.md [hunk 5]
```diff
@@ -99,11 +99,72 @@ If you use a custom webhook for your destination and need to embed JSON in the m
...
+      "throttle": {
+        "value": 27,
+        "unit": "MINUTES"
+      },
+      "subject_template": {
+        "source": "TheSubject"
+      }
+    }]
+  }],
   "rbac_roles": ["role1", "role2"]
```
Comment (lezzago) [inline _monitoring-plugins/alerting/api.md:168]: Is it possible to highlight this line further?

File: _monitoring-plugins/alerting/security.md [hunk 6]
```diff
@@ -91,19 +91,21 @@ In a create monitor scenario, follow these guidelines to specify roles:
...
-Admin user | User-specified | Remove all the backend roles associate to the monitor and then use all the specified backend roles to associate to the monitor.
-Regular user | Yes | Remove backend roles associated to the monitor that the user has access to, but didn’t specify. Then add all the other specified backend roles (that the user already has in their list of backend roles) to the monitor.
+Admin user | Yes | Remove all the backend roles associate to the monitor and then use all the specified backend roles to associate to the monitor.
+Regular user | Yes | Remove backend roles associated to the monitor that the user has access to, but didn’t specify. Then add all the other specified backend roles from the list of backend roles that the user has permission to use to the monitor.
 Regular user | No | Don’t update the backend roles on the monitor.
 
-Note that an empty list is considered the same as removing all permissions that the user possesses. Also note that if the user tries to associate roles that they don't have permission to use, it will throw an exception.
+- For admin users, an empty list is considered the same as removing all permissions that the user possesses.
```
Comment (lezzago) [inline _monitoring-plugins/alerting/security.md:106]: We need to state if a non-admin user passes in an empty list, that will throw an exception as that behavior is not allow by non-admin users.

**Review by lezzago — CHANGES_REQUESTED** [review]

File: _monitoring-plugins/alerting/api.md (hunk 5 above)
Comment (alicejw-aws) [reply _monitoring-plugins/alerting/api.md:168]: i don't think we can put bold font within the code block. let me emphasize it in the text above.

**Comment by alicejw-aws** [comment]
Hi @lezzago , i've made the updates. Please take a look and verify. thanks

**Review by lezzago — APPROVED** [review]

File: _monitoring-plugins/alerting/api.md [hunk 7]
```diff
@@ -99,6 +99,78 @@ If you use a custom webhook for your destination and need to embed JSON in the m
...
+      "subject_template": {
+        "source": "TheSubject"
+      }
+    }]
+  }],
+  "rbac_roles": ["role1", "role2"]
+}
+```
+
+To learn more about using backend roles to limit access, see [\(Advanced\) Limit access by backend role]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/security/#advanced-limit-access-by-backend-role).
```
Comment (Naarcha-AWS) [inline _monitoring-plugins/alerting/api.md:172]: Are the backslashes here intentional?

File: _monitoring-plugins/alerting/security.md [hunk 8]
```diff
@@ -82,3 +82,42 @@ If you only want users to be able to see and modify their own monitors and desti
...
+
+User type  | Role is specified by user or not (Y/N) | How to use the RBAC roles
+:--- | :--- | :---
+Admin user | Yes | Remove all the backend roles associate to the monitor and then use all the specified backend roles to associate to the monitor.
+Regular user | Yes | Remove backend roles associated to the monitor that the user has access to, but didn’t specify. Then add all the other specified backend roles from the list of backend roles that the user has permission to use to the monitor.
+Regular user | No | Don’t update the backend roles on the monitor.
+
+- For admin users, an empty list is considered the same as removing all permissions that the user possesses. If a non-admin user passes in an empty list, that will throw an exception, because that is not allowed by non-admin users.
+- If the user tries to associate roles that they don't have permission to use, it will throw an exception.
+{: .note }
```
Comment (Naarcha-AWS) [inline _monitoring-plugins/alerting/security.md:108]: Does the above unordered list format correctly in your local jekyll build?

**Review by Naarcha-AWS — APPROVED** [review]
A couple of short comments, otherwise LGTM.

File: _monitoring-plugins/alerting/api.md [hunk 9]
```diff
@@ -99,6 +99,78 @@ If you use a custom webhook for your destination and need to embed JSON in the m
 }
 ```
 
+Optionally, to specify a backend role, you can add the `rbac_roles` parameter and backend role names to the bottom of your create monitor request.
+
+#### Sample request
+
+The following request creates a query-level monitor and provides two backend roles `role1` and `role2`. The section at the bottom of the request shows the line that specifies the roles with this syntax: `"rbac_roles": ["role1", "role2"]`.
```
Comment (cwillum) [inline _monitoring-plugins/alerting/api.md:106]: "... and provides two backend roles, `role1` and `role2`."
just a comma.

File: _monitoring-plugins/alerting/security.md [hunk 10]
```diff
@@ -82,3 +82,42 @@ If you only want users to be able to see and modify their own monitors and desti
 ```
 
 Then, use this new role for all alerting users. -->
+
+### Specify RBAC backend roles
+
+You can specify RBAC backend roles when you create or update a monitor with the Alerting API.
```
Comment (cwillum) [inline _monitoring-plugins/alerting/security.md:88]: might be worthwhile spelling out Rule Based Access Control here. I don't think it appears before this on the page.

File: _monitoring-plugins/alerting/security.md [hunk 11]
```diff
@@ -82,3 +82,42 @@ If you only want users to be able to see and modify their own monitors and desti
...
+:--- | :--- | :---
+Admin user | Yes | Use all the specified backend roles to associate to the monitor.
+Regular user | Yes | Use all the specified backend roles from the list of backend roles that the user has permission to use to associate with the monitor.
+Regular user | No | Copy user’s backend roles and associate them to the monitor.
+
+In an update monitor scenario, follow these guidelines to specify roles:
+
+User type  | Role is specified by user or not (Y/N) | How to use the RBAC roles
+:--- | :--- | :---
+Admin user | Yes | Remove all the backend roles associate to the monitor and then use all the specified backend roles to associate to the monitor.
```
Comment (cwillum) [inline _monitoring-plugins/alerting/security.md:102]: "Remove all the backend roles associated to the monitor ..."
"associated" correct?

**Review by cwillum — APPROVED** [review]
LGTM, with suggestions.

File: _monitoring-plugins/alerting/security.md (hunk 10 above)
Comment (alicejw-aws) [reply _monitoring-plugins/alerting/security.md:88]: thanks. adding "role-based access control" with acronym in parentheses

File: _monitoring-plugins/alerting/security.md (hunk 8 above)
Comment (alicejw-aws) [reply _monitoring-plugins/alerting/security.md:108]: yes. i used asterisks before, but got MDLint error: expected dashes. 
the bullets render within the callout like this:
Screen Shot 2022-11-15 at 9 30 45 AM

File: _monitoring-plugins/alerting/api.md (hunk 7 above)
Comment (alicejw-aws) [reply _monitoring-plugins/alerting/api.md:172]: i thought it wouldn't render the link properly with the title in parenthesis. but tested it without, and it's fine. thanks!

**Pull request merged**